### DISCUSSION TITLE
budget allocation with no control vars

### DISCUSSION THREAD

**Discussion by bella0715** [pymc-labs/pymc-marketing#1028] [Q&A]
Here's how I built the model.
```
mmm = DelayedSaturatedMMM(
    model_config = mmm_config,
    sampler_config = sampler_config,
    date_column = date_var,
    channel_columns = spend_vars,
    control_columns = None,
    adstock_max_lag=8,
    yearly_seasonality=1,
)

```

When I try to run the below code to do the budget allocation, I get an error 'UnboundLocalError: cannot access local variable '_controls' where it is not associated with a value'. How do I run the budget allocation without control_columns?

```
response = mmm.allocate_budget_to_maximize_response(
    budget=total_budget,
    num_days=8,
    time_granularity="weekly",
    budget_bounds=budget_bounds,
)
```

**Comment by williambdean** [comment]
Sounds like a bug. Items like this can be raised as issues instead of discussions. We can continue there
